router/tree.go

```diff
@@ -180,7 +181,6 @@ func (t *spanningTree) becomeRoot() {
 	if !t.Coords().EqualTo(newCoords) {
 		go t.callback(0, types.SwitchPorts{})
 	}
-	t.advertise()
```

Comment:
In the case of losing the node's parent connection then becoming root itself, won't removing this potentially lead to a long wait time before this node sends out a root announcement? (depending on where the announcement worker ticker is currently at)
Or am I missing something...

Reply:
This was mostly just a case of moving where `t.advertise()` was called, but actually it probably makes more sense to move where `t.becomeParent()` is called instead (e.g. not right at the beginning of `selectNewParent()`).

Reply:
Ah I see. Looks good now :+1: